File: PawNClaw.Staff.Mobile/lib/models/booking.dart
import 'booking_detail.dart';
import 'pet.dart';
import 'package:intl/intl.dart';

class Booking {
  int? id;
  // int? subTotal;
  // Null? discount;
  // int? total;
  // Null? checkIn;
  // Null? checkOut;
  // String? createTime;
  DateTime? startBooking;
  DateTime? endBooking;
  // int? statusId;
  // Null? voucherCode;
  // int? customerId;
  // int? centerId;
  // Null? rating;
  // Null? customerNote;
  // String? staffNote;
  // Null? center;
  Customer? customer;
  double? totalSupply;
  double? totalService;
  // Null? status;
  // Null? voucherCodeNavigation;
  // Null? generalLedger;
  List<BookingDetails>? bookingDetails;
  // List<Null>? petHealthHistories;
  // List<Null>? serviceOrders;
  // List<Null>? supplyOrders;

  Booking({
    this.id,
    // this.subTotal,
    // this.discount,
    // this.total,
    // this.checkIn,
    // this.checkOut,
    // this.createTime,
    this.startBooking,
    this.endBooking,
    // this.statusId,
    // this.voucherCode,
    // this.customerId,
    // this.centerId,
    // this.rating,
    // this.customerNote,
    // this.staffNote,
    // this.center,
    this.customer,
    this.totalSupply,
    this.totalService,
    // this.status,
    // this.voucherCodeNavigation,
    // this.generalLedger,
    this.bookingDetails,
    // this.petHealthHistories,
    // this.serviceOrders,
    // this.supplyOrders
  });

  Booking.fromJson(Map<String, dynamic> json) {
    id = json['id'];
    // subTotal = json['subTotal'];
    // discount = json['discount'];
    // total = json['total'];
    // checkIn = json['checkIn'];
    // checkOut = json['checkOut'];
    // createTime = json['createTime'];
    startBooking = DateTime.parse(json['startBooking']);
    endBooking = DateTime.parse(json['endBooking']);
    // statusId = json['statusId'];
    // voucherCode = json['voucherCode'];
    // customerId = json['customerId'];
    // centerId = json['centerId'];
    // rating = json['rating'];
    // customerNote = json['customerNote'];
    // staffNote = json['staffNote'];
    // center = json['center'];
    customer = json['customer'] != null
        ? new Customer.fromJson(json['customer'])
        : null;
    totalSupply = json['totalSupply'];
    totalService = json['totalService'];
    // status = json['status'];
    // voucherCodeNavigation = json['voucherCodeNavigation'];
    // generalLedger = json['generalLedger'];
    if (json['bookingDetails'] != null) {
      bookingDetails = <BookingDetails>[];
      json['bookingDetails'].forEach((v) {
        bookingDetails!.add(new BookingDetails.fromJson(v));
      });
    }
    // if (json['petHealthHistories'] != null) {
    //   petHealthHistories = <Null>[];
    //   json['petHealthHistories'].forEach((v) {
    //     petHealthHistories!.add(new Null.fromJson(v));
    //   });
    // }
    // if (json['serviceOrders'] != null) {
    //   serviceOrders = <Null>[];
    //   json['serviceOrders'].forEach((v) {
    //     serviceOrders!.add(new Null.fromJson(v));
    //   });
    // }
    // if (json['supplyOrders'] != null) {
    //   supplyOrders = <Null>[];
    //   json['supplyOrders'].forEach((v) {
    //     supplyOrders!.add(new Null.fromJson(v));
    //   });
    // }
  }

  Map<String, dynamic> toJson() {
    final Map<String, dynamic> data = new Map<String, dynamic>();
    data['id'] = this.id;
    // data['subTotal'] = this.subTotal;
    // data['discount'] = this.discount;
    // data['total'] = this.total;
    // data['checkIn'] = this.checkIn;
    // data['checkOut'] = this.checkOut;
    // data['createTime'] = this.createTime;
    data['startBooking'] =
        DateFormat('yyyy-MM-dd HH:mm:ss').format(this.startBooking!);
    data['endBooking'] =
        DateFormat('yyyy-MM-dd HH:mm:ss').format(this.endBooking!);
    // data['statusId'] = this.statusId;
    // data['voucherCode'] = this.voucherCode;
    // data['customerId'] = this.customerId;
    // data['centerId'] = this.centerId;
    // data['rating'] = this.rating;
    // data['customerNote'] = this.customerNote;
    // data['staffNote'] = this.staffNote;
    // data['center'] = this.center;
    if (this.customer != null) {
      data['customer'] = this.customer!.toJson();
    }
    // data['status'] = this.status;
    // data['voucherCodeNavigation'] = this.voucherCodeNavigation;
    // data['generalLedger'] = this.generalLedger;
    if (this.bookingDetails != null) {
      data['bookingDetails'] =
          this.bookingDetails!.map((v) => v.toJson()).toList();
    }
    // if (this.petHealthHistories != null) {
    //   data['petHealthHistories'] =
    //       this.petHealthHistories!.map((v) => v.toJson()).toList();
    // }
    // if (this.serviceOrders != null) {
    //   data['serviceOrders'] =
    //       this.serviceOrders!.map((v) => v.toJson()).toList();
    // }
    // if (this.supplyOrders != null) {
    //   data['supplyOrders'] = this.supplyOrders!.map((v) => v.toJson()).toList();
    // }
    return data;
  }
}

class Customer {
  int? id;
  String? name;
  // String? birth;
  // Null? gender;
  // Null? idNavigation;
  // List<Null>? bookings;
  // List<Null>? customerAddresses;
  // List<Null>? pets;

  Customer({
    this.id,
    this.name,
    // this.birth,
    // this.gender,
    // this.idNavigation,
    // this.bookings,
    // this.customerAddresses,
    // this.pets
  });

  Customer.fromJson(Map<String, dynamic> json) {
    id = json['id'];
    name = json['name'];
    // birth = json['birth'];
    // gender = json['gender'];
    // idNavigation = json['idNavigation'];
    // if (json['bookings'] != null) {
    //   bookings = <Null>[];
    //   json['bookings'].forEach((v) {
    //     bookings!.add(new Null.fromJson(v));
    //   });
    // }
    // if (json['customerAddresses'] != null) {
    //   customerAddresses = <Null>[];
    //   json['customerAddresses'].forEach((v) {
    //     customerAddresses!.add(new Null.fromJson(v));
    //   });
    // }
    // if (json['pets'] != null) {
    //   pets = <Null>[];
    //   json['pets'].forEach((v) {
    //     pets!.add(new Null.fromJson(v));
    //   });
    // }
  }

  Map<String, dynamic> toJson() {
    final Map<String, dynamic> data = new Map<String, dynamic>();
    data['id'] = this.id;
    data['name'] = this.name;
    // data['birth'] = this.birth;
    // data['gender'] = this.gender;
    // data['idNavigation'] = this.idNavigation;
    // if (this.bookings != null) {
    //   data['bookings'] = this.bookings!.map((v) => v.toJson()).toList();
    // }
    // if (this.customerAddresses != null) {
    //   data['customerAddresses'] =
    //       this.customerAddresses!.map((v) => v.toJson()).toList();
    // }
    // if (this.pets != null) {
    //   data['pets'] = this.pets!.map((v) => v.toJson()).toList();
    // }
    return data;
  }
}

File: PawNClaw.Backend/PawNClaw.Data/Database/Booking.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using Microsoft.EntityFrameworkCore;

#nullable disable

namespace PawNClaw.Data.Database
{
    public partial class Booking
    {
        public Booking()
        {
            BookingActivities = new HashSet<BookingActivity>();
            BookingDetails = new HashSet<BookingDetail>();
            CancelLogs = new HashSet<CancelLog>();
            PetHealthHistories = new HashSet<PetHealthHistory>();
            ServiceOrders = new HashSet<ServiceOrder>();
            SupplyOrders = new HashSet<SupplyOrder>();
        }

        [Key]
        [Column("id")]
        public int Id { get; set; }
        [Column("sub_total", TypeName = "numeric(19, 5)")]
        public decimal? SubTotal { get; set; }
        [Column("discount", TypeName = "numeric(19, 5)")]
        public decimal? Discount { get; set; }
        [Column("total", TypeName = "numeric(19, 5)")]
        public decimal? Total { get; set; }
        [Column("check_in", TypeName = "datetime")]
        public DateTime? CheckIn { get; set; }
        [Column("check_out", TypeName = "datetime")]
        public DateTime? CheckOut { get; set; }
        [Column("create_time", TypeName = "datetime")]
        public DateTime? CreateTime { get; set; }
        [Column("start_booking", TypeName = "datetime")]
        public DateTime? StartBooking { get; set; }
        [Column("end_booking", TypeName = "datetime")]
        public DateTime? EndBooking { get; set; }
        [Column("status_id")]
        public int StatusId { get; set; }
        [Column("voucher_code")]
        [StringLength(32)]
        public string VoucherCode { get; set; }
        [Column("customer_id")]
        public int CustomerId { get; set; }
        [Column("center_id")]
        public int CenterId { get; set; }
        [Column("rating")]
        public byte? Rating { get; set; }
        [Column("customer_note")]
        [StringLength(512)]
        public string CustomerNote { get; set; }
        [Column("staff_note")]
        [StringLength(512)]
        public string StaffNote { get; set; }
        [Column("invoice_url")]
        [StringLength(1024)]
        public string InvoiceUrl { get; set; }
        [Column("feedback")]
        [StringLength(1024)]
        public string Feedback { get; set; }

        [NotMapped]
        public decimal? TotalSupply { get; set; }
        [NotMapped]
        public decimal? TotalService { get; set; }
        [NotMapped]
        public decimal? TotalCage { get; set; }

        [ForeignKey(nameof(CenterId))]
        [InverseProperty(nameof(PetCenter.Bookings))]
        public virtual PetCenter Center { get; set; }
        [ForeignKey(nameof(CustomerId))]
        [InverseProperty("Bookings")]
        public virtual Customer Customer { get; set; }
        [ForeignKey(nameof(StatusId))]
        [InverseProperty(nameof(BookingStatus.Bookings))]
        public virtual BookingStatus Status { get; set; }
        [ForeignKey(nameof(VoucherCode))]
        [InverseProperty(nameof(Voucher.Bookings))]
        public virtual Voucher VoucherCodeNavigation { get; set; }
        [InverseProperty("IdNavigation")]
        public virtual GeneralLedger GeneralLedger { get; set; }
        [InverseProperty(nameof(BookingActivity.Booking))]
        public virtual ICollection<BookingActivity> BookingActivities { get; set; }
        [InverseProperty(nameof(BookingDetail.Booking))]
        public virtual ICollection<BookingDetail> BookingDetails { get; set; }
        [InverseProperty(nameof(CancelLog.Booking))]
        public virtual ICollection<CancelLog> CancelLogs { get; set; }
        [InverseProperty(nameof(PetHealthHistory.Booking))]
        public virtual ICollection<PetHealthHistory> PetHealthHistories { get; set; }
        [InverseProperty(nameof(ServiceOrder.Booking))]
        public virtual ICollection<ServiceOrder> ServiceOrders { get; set; }
        [InverseProperty(nameof(SupplyOrder.Booking))]
        public virtual ICollection<SupplyOrder> SupplyOrders { get; set; }

        [NotMapped]
        public ICollection<Photo> Photos { get; set; }
    }
}
